src/InterestPayout.Worker/WebApi/AssetInterestsController.cs

```diff
@@ -30,45 +31,35 @@ public class AssetInterestsController : ControllerBase
 
         [HttpPost("create-or-update")]
         public async Task<ActionResult> CreateOrUpdate(
-            [FromBody] AssetInterestCreateOrUpdateRequest createOrUpdateRequest)
+            [FromBody] AssetInterestCreateOrUpdateRequest request,
+            [Required, FromHeader(Name = "X-Idempotency-ID")] string idempotencyId)
         {
-            if (createOrUpdateRequest == null)
+            if (request == null)
                 return BadRequest("Request is required.");
-            if(string.IsNullOrWhiteSpace(createOrUpdateRequest.AssetId))
+            if(string.IsNullOrWhiteSpace(request.AssetId))
                 return BadRequest("AssetId is required.");
-            if (createOrUpdateRequest.InterestRate < 100m)
+            if (request.InterestRate < 100m)
```

Comment:
No, I think it is fine as it is.  150% interest is a valid case, but -150% is not valid, since you only have 100% of what you have (if that makes sense)

Reply:
but current check doesn't allow you to specify any negative values

Reply:
sorry, I misunderstood oroginally, fixed